lib/web/ui/usercontext.go

```diff
@@ -207,9 +207,22 @@ func NewUserContext(user types.User, userRoles services.RoleSet, features proto.
 	dbServerAccess := newAccess(userRoles, ctx, types.KindDatabaseServer)
 	dbAccess := newAccess(userRoles, ctx, types.KindDatabase)
 	kubeServerAccess := newAccess(userRoles, ctx, types.KindKubeServer)
-	requestAccess := newAccess(userRoles, ctx, types.KindAccessRequest)
 	desktopAccess := newAccess(userRoles, ctx, types.KindWindowsDesktop)
 	cnDiagnosticAccess := newAccess(userRoles, ctx, types.KindConnectionDiagnostic)
+	requestAccess := newAccess(userRoles, ctx, types.KindAccessRequest)
+
+	// If the user has a requestable role or any `search_as_roles`,
+	// they can create access requests for themselves even without an
+	// explicit `allow` rule
+	if len(userRoles.GetAllowedSearchAsRoles()) > 0 || len(accessCapabilities.RequestableRoles) > 0 {
```

Comment:
what about reviewers who cannot themselves create access requests, they should also be able to see the page

Reply:
Thanks, added the following block to allow reviewers to see the screens:
```go
	// If the user has preview_requests or preview_as_roles,
	// they should be able to see the Access Requests screens in order to review requests
	if len(userRoles.GetAllowedPreviewAsRoles()) > 0 || userRoles.WithoutImplicit().MaybeCanReviewRequests() {
		requestAccess = access{
			List:   true,
			Read:   true,
			Create: true,
			Edit:   true,
			Delete: true,
		}
	}
```